### PR TITLE
Notebooks docs

### PR DESCRIPTION
**Pull request by benvenker**
**Please don't merge yet! Liberal comments welcome.**
The goal of this PR is just to get a stake in the ground for Notebooks docs so we can iteratively improve them. I'm taking a page from Code Insights and not adding Notebooks to the main nav on docs.sourcegraph while in Beta.

### TODOs
- [ ] Upload videos and images to GCP

### CODE REVIEW

**Comment by benvenker** [comment]
@novoselrok i don't currently have access to GCP and was curious if you'd mind uploading the files for me. If it's a huge hassle I'll see if I can get my own account.

**Comment by benvenker** [comment]
@novoselrok I think we're good now! Just added sharing this morning.

**Comment by benvenker** [comment]
Ah, actually I need to put in how to enable in on-prem. I'll ping you when that's done @novoselrok

**Comment by benvenker** [comment]
@jyavorska happy to have you look at this as well but pinging you because you're the resident handbook guru and buildkite keeps failing on me but I can't see "details" so don't know why. Any ideas on what can cause that?

**Comment by jyavorska** [comment]
Sorry @benvenker I'm no expert on the docs site. I don't even seem to have access to the build logs.

**Comment by benvenker** [comment]
Dang. ok thanks!

**Comment by benvenker** [comment]
@novoselrok this is ready for review. I'm going to add snippets documentation in another RP.

File: doc/notebooks/index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<style>
```
Comment (novoselrok) [inline doc/notebooks/index.md:1]: Why is this style tag necessary?

File: doc/notebooks/index.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+</style>
+
+# Notebooks
+Notebooks enable powerful live–and persistent–documentation, shareable with your organization or the world. Notebooks are currently in public [Beta](https://sourcegraph.com/notebooks?order=stars-desc) on Sourcegraph Cloud and in Sourcegraph enterprise installs at version 3.36 or later. You can explore all the public notebooks on Sourcegraph.com without an account, or create a [Sourcegraph Cloud](https://about.sourcegraph.com/get-started/cloud) account to start creating your own Notebooks.
+
+Inspired by Jupyter Notebooks and powered by Markdown and Sourcegraph's code search, Notebooks let you and your team create living documentation that interacts directly with your code. You can leverage Notebooks to onboard a new teammate, document [known vulnerabilities](https://sourcegraph.com/notebooks/Tm90ZWJvb2s6MQ==), a [common pattern](https://sourcegraph.com/notebooks/Tm90ZWJvb2s6OTI=) in your codebase, or [useful Sourcegraph queries](https://sourcegraph.com/notebooks/Tm90ZWJvb2s6MTU=).
+
+<!-- Notebooks image TODO: get uploaded to GCP -->
```
Comment (novoselrok) [inline doc/notebooks/index.md:39]: You can probably remove the todo comment.

**Review by novoselrok — APPROVED** [review]

**Pull request merged**